### PR TITLE
chore: add circuit breaker to phase2 gov script

### PR DESCRIPTION
**Pull request by ninocomputer**
### Description

Configure circuit breaker 

New pool configs were pulled from nadiem's working branch. 🚧 do not merge before [his pr](https://github.com/mento-protocol/mento-deployment/pull/67). 🚧

### Other changes

n/a

### Tested

n/a

### Related issues

-  https://github.com/mento-protocol/mento-deployment/issues/58
- Closes https://github.com/mento-protocol/mento-deployment/issues/66

### Backwards compatibility

n/a

### Documentation

n/a

### CODE REVIEW

File: script/upgrades/MU01/governance/Config.sol
Comment (ninocomputer) [inline script/upgrades/MU01/governance/Config.sol:374]: do not review pool configs in this pr

File: script/upgrades/MU01/governance/MU01-CGP-Phase2.sol
Comment (philbow61) [inline script/upgrades/MU01/governance/MU01-CGP-Phase2.sol:288]: I think this is not doing the correct thing:
 we want:
 ` setRateFeedDependencies(cEURUSDCConfig.referenceRateFeedID, [cUSDUSDCConfig.referenceRateFeedID])`
and 
` setRateFeedDependencies(cBRLUSDCConfig.referenceRateFeedID, [cUSDUSDCConfig.referenceRateFeedID])`
based on the breaker box function:
`function setRateFeedDependencies(address rateFeedID, address[] memory dependencies)`
right?

File: script/upgrades/MU01/governance/MU01-CGP-Phase2.sol
Comment (philbow61) [inline script/upgrades/MU01/governance/MU01-CGP-Phase2.sol:314]: I am not sure whether we want the value delta breaker for all rateFeeds in phase 1 we only enabled it for cUSD/USDC

**Review by philbow61 — CHANGES_REQUESTED** [review]

File: script/upgrades/MU01/governance/MU01-CGP-Phase2.sol
Comment (ninocomputer) [reply script/upgrades/MU01/governance/MU01-CGP-Phase2.sol:266]: yeah, my bad didn't double check that in the new setup trading halted is 3. :)

File: script/upgrades/MU01/governance/MU01-CGP-Phase2.sol
Comment (philbow61) [inline script/upgrades/MU01/governance/MU01-CGP-Phase2.sol:398]: i think this should be the rateFeedID and not the whole pool config struct

**Review by philbow61 — CHANGES_REQUESTED** [review]

File: script/upgrades/MU01/governance/MU01-CGP-Phase2.sol
Comment (ninocomputer) [reply script/upgrades/MU01/governance/MU01-CGP-Phase2.sol:398]: correct

File: script/upgrades/MU01/governance/MU01-CGP-Phase2.sol
Comment (ninocomputer) [reply script/upgrades/MU01/governance/MU01-CGP-Phase2.sol]: I think input validation is good but I'll add that check on the side 👍🏼

**Review by philbow61 — APPROVED** [review]

File: script/upgrades/MU01/governance/MU01-CGP-Phase2.sol
Comment (bayological) [inline script/upgrades/MU01/governance/MU01-CGP-Phase2.sol:259]: We need to add all rate feed ids, including the new ones for EUR & BRL, to the mapping of monitored rate feed ids in the breaker box.
This was previously done in the initializer that was removed. This is done by calling the function `addRateFeeds`

File: script/upgrades/MU01/governance/Config.sol
Comment (bayological) [inline script/upgrades/MU01/governance/Config.sol:391]: @sissnad Just double checking this is intentional. Buth new pools(USDCEUR & USDCBRL) are using the value delta breaker instead of the median delta breaker.

File: script/upgrades/MU01/governance/MU01-CGP-Phase2.sol
Comment (bayological) [inline script/upgrades/MU01/governance/MU01-CGP-Phase2.sol:354]: We're missing the configuration of the value delta breaker for the new pools that were added. The new config for both of these pools is set to use this breaker. If Nadiem confirms that they should be using the value delta breaker, this should be added.

File: script/upgrades/MU01/governance/Config.sol
Comment (bayological) [inline script/upgrades/MU01/governance/Config.sol:1]: The acceptance criteria for this issue require the simulation tests to run and pass. Were these checked?

**Review by bayological — CHANGES_REQUESTED** [review]

File: script/upgrades/MU01/governance/Config.sol
Comment (ninocomputer) [reply script/upgrades/MU01/governance/Config.sol:374]: they should be reviewed in Nadiem's pr. There was no way for me to work on this unless I pulled Nadiem's branch. I've stated in the description to not merge this before his pr is merged and I'd pull the latest so there'd be no diff after. I hope this clears up the confusion.

File: script/upgrades/MU01/governance/MU01-CGP-Phase2.sol
Comment (ninocomputer) [reply script/upgrades/MU01/governance/MU01-CGP-Phase2.sol:101]: This is TBD. It's the first time we're deploying cEUR/USDC, cBRL/USDC that's why they are in phase1

File: script/upgrades/MU01/governance/Config.sol
Comment (ninocomputer) [reply script/upgrades/MU01/governance/Config.sol:321]: I set them to 0 to make it seem like they weren't configured because otherwise I wasn't able to compile the script

File: script/upgrades/MU01/governance/Config.sol
Comment (ninocomputer) [reply script/upgrades/MU01/governance/Config.sol:1]: they were not checked, since the config needs to be finished first.

File: script/upgrades/MU01/governance/MU01-CGP-Phase2.sol
Comment (ninocomputer) [reply script/upgrades/MU01/governance/MU01-CGP-Phase2.sol:354]: I think new pools will be configured with medianDeltaBreaker

File: script/upgrades/MU01/governance/Config.sol
Comment (ninocomputer) [reply script/upgrades/MU01/governance/Config.sol:391]: We will update this when the config's ready

File: script/upgrades/MU01/governance/MU01-CGP-Phase2.sol
Comment (ninocomputer) [reply script/upgrades/MU01/governance/MU01-CGP-Phase2.sol:259]: 👍🏼

File: script/upgrades/MU01/governance/Config.sol
Comment (ninocomputer) [reply script/upgrades/MU01/governance/Config.sol:430]: This was pulled from draft pr and will be updated to correct value once config's done

File: script/upgrades/MU01/governance/Config.sol
Comment (ninocomputer) [reply script/upgrades/MU01/governance/Config.sol:395]: ditto

File: script/upgrades/MU01/governance/Config.sol
Comment (ninocomputer) [reply script/upgrades/MU01/governance/Config.sol:386]: 👍🏼

**Pull request closed without merging**
